src/components/dialogs/NameRequestInvalidErrorDialog.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <v-dialog v-model="dialog" width="45rem" persistent :attach="attach" content-class="namerequest-invalid-error-dialog">
+    <v-card>
+      <v-card-title id="dialog-title" v-if="type === NameRequestStates.NOTFOUND">Name Request Not Found</v-card-title>
+      <v-card-title id="dialog-title" v-else>Name Request Invalid</v-card-title>
+      <v-card-text>
+        <p class="genErr" v-if="type === NameRequestStates.EXPIRED">
+          The specified name request has expired.</p>
+
+        <p class="genErr" v-else-if="type === NameRequestStates.CONSUMED">
+          The specified name request has already been consumed.</p>
+
+        <p class="genErr" v-else-if="type === NameRequestStates.NOTAPPROVED">
+          The specified name request has not been approved.</p>
+
+        <p class="genErr" v-else-if="type === NameRequestStates.NOTFOUND">
+          The specified name request number could not be found.</p>
+
+        <p class="genErr" v-else>An unexpected error has occurred.</p>
+      </v-card-text>
+
+      <v-divider></v-divider>
+
+      <v-card-actions>
+        <v-btn v-if="type === NameRequestStates.NOTFOUND || type === NameRequestStates.CONSUMED"
+                id="dialog-redirect-button" color="primary" text @click="redirect()">Back to Dashboard</v-btn>
+
+        <v-btn v-else id="dialog-ok-button" color="primary" text @click="okay()">OK</v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Emit } from 'vue-property-decorator'
+import { NameRequestStates } from '@/enums/nameRequestStates'
+
+@Component({
+  computed: {}
+})
+export default class NameRequestInvalidErrorDialog extends Vue {
+  // NameRequestStates Enum
+  readonly NameRequestStates: {} = NameRequestStates
```

Comment:
AS of TS 3.8.3  we don't need to assign a type when initializing readonly properties ( or so i've been told! ) Try it out, if no type errors than perhaps do away with the `: {}` ?

Reply:
updated!